typescript/sdk/src/providers/SmartProvider/SmartProvider.foundry-test.ts

```diff
@@ -156,6 +158,7 @@ describe('SmartProvider', async () => {
     try {
       await token.transfer(constants.AddressZero, 1000000);
     } catch (e: any) {
+      expect(e.error.cause.code).to.equal('UNPREDICTABLE_GAS_LIMIT');
```

Comment:
Maybe to avoid magic strings, we could have an enum to encode the errors from the smart provider that we are inteerested into

Reply:
good point, can import from ethers